src/WinRT.Runtime/Marshalers.cs

```diff
@@ -360,23 +360,24 @@ public class MarshalGeneric<T>
 
         static MarshalGeneric()
         {
-            CreateMarshaler = BindCreateMarshaler();
-            GetAbi = BindGetAbi();
-            FromAbi = BindFromAbi();
-            CopyAbi = BindCopyAbi();
-            FromManaged = BindFromManaged();
-            CopyManaged = BindCopyManaged();
-            DisposeMarshaler = BindDisposeMarshaler();
-            DisposeAbi = BindDisposeAbi();
-            CreateMarshalerArray = BindCreateMarshalerArray();
-            GetAbiArray = BindGetAbiArray();
-            FromAbiArray = BindFromAbiArray();
-            FromManagedArray = BindFromManagedArray();
-            DisposeMarshalerArray = BindDisposeMarshalerArray();
-            DisposeAbiArray = BindDisposeAbiArray();
+            CreateMarshaler = (T value) => CreateMarshalerLazy.Value(value);
+            GetAbi = (object objRef) => GetAbiLazy.Value(objRef);
+            FromAbi = (object box) => FromAbiLazy.Value(box);
+            CopyAbi = (object box, IntPtr dest) => CopyAbiLazy.Value(box, dest);
+            FromManaged = (T value) => FromManagedLazy.Value(value);
+            CopyManaged = (T value, IntPtr dest) => CopyManagedLazy.Value(value, dest);
+            DisposeMarshaler = (object objRef) => DisposeMarshalerLazy.Value(objRef);
+            DisposeAbi = (object box) => DisposeAbiLazy.Value(box);
+            CreateMarshalerArray = (T[] array) => CreateMarshalerArrayLazy.Value(array);
+            GetAbiArray = (object box) => GetAbiArrayLazy.Value(box);
+            FromAbiArray = (object box) => FromAbiArrayLazy.Value(box);
+            FromManagedArray = (T[] array) => FromManagedArrayLazy.Value(array);
+            DisposeMarshalerArray = (object box) => DisposeMarshalerArrayLazy.Value(box);
+            DisposeAbiArray = (object box) => DisposeAbiArrayLazy.Value(box);
         }
 
         public static readonly Func<T, object> CreateMarshaler;

```

Comment:
```suggestion
        public static readonly Func<T, object> CreateMarshaler = (T value) => CreateMarshalerLazy.Value(value);
```
It is better to initialize statics inline like this. It creates before-field-init constructor that is cheaper than the explicit constructor that you have there now.

Change all fields to use this pattern and delete `static MarshalGeneric()` above.

Reply:
Thanks addressed.